cmd/gather.go

```diff
@@ -35,13 +36,26 @@ var gatherCmd = &cobra.Command{
 			os.Exit(1)
 		}
 
+		if argStatic && argDynamic {
+			log.Fatalf("Cannot use both 'static' and 'dynamic' only gatherers - exiting")
+			os.Exit(1)
+		}
+
+		gathererTypeArg := ""
+		if argStatic {
+			gathererTypeArg = "static"
+		}
+		if argDynamic {
+			gathererTypeArg = "dynamic"
+		}
+
 		//run the gatherers...
 		gathererSlice := gatherers.GetGatherers()
 
 		var facts []gatherers.GatheredFact
 
 		for _, e := range gathererSlice {
-			if e.AppliesToEnvironment() {
+			if (e.GetGathererCmdType() == gathererTypeArg || gathererTypeArg == "") && e.AppliesToEnvironment() {
```

Comment:
Two things - First, I think if we set a default gatherer type (say "static") - we can avoid the "||" here.

Second - I think this relies on the assumption of short circuit evaluation of boolean operators. I think it would actually be better if we moved the "e.AppliedToEnvironment()" into a nested "if" here. We actually want it to be explicit that we _only run the "AppliesToEnvironment" if this is the right gatherer type_

Reply:
1. Removed the empty type conditional 2. Moved AppliesToEnvironment check and GetGathererCmdType check into separate conditionals